src/hooks/useAllowance.ts

```diff
@@ -8,70 +8,76 @@ import { useBlockNumber } from 'state/block-number/hooks';
 import useCurrentNetwork from './useCurrentNetwork';
 import useWalletService from './useWalletService';
 
-function useAllowance(
-  from: Token | undefined | null
-): [{ token: Token; allowance: string } | undefined, boolean, string?] {
-  const [isLoading, setIsLoading] = React.useState(false);
+type Allowance = {
+  token: Token;
+  allowance: string | undefined;
+};
+
+type AllowanceResponse = [Allowance, boolean, string?];
+
+const dummyToken: Allowance = { token: EMPTY_TOKEN, allowance: undefined };
+
+function useAllowance(from: Token | undefined | null): AllowanceResponse {
   const walletService = useWalletService();
-  const [result, setResult] = React.useState<{ token: Token; allowance: string } | undefined>(undefined);
-  const [error, setError] = React.useState<string | undefined>(undefined);
+  const [state, setState] = React.useState({ isLoading: false, result: dummyToken, error: undefined });
   const hasPendingTransactions = useHasPendingTransactions();
   const prevFrom = usePrevious(from);
   const prevPendingTrans = usePrevious(hasPendingTransactions);
-  const account = usePrevious(walletService.getAccount());
+  const prevAccount = usePrevious(walletService.getAccount());
+  const account = walletService.getAccount();
   const currentNetwork = useCurrentNetwork();
   const blockNumber = useBlockNumber(currentNetwork.chainId);
   const prevBlockNumber = usePrevious(blockNumber);
-  const prevResult = usePrevious(result, false);
+  const prevResult = usePrevious(state.result, false, 'allowance');
 
   React.useEffect(() => {
     async function callPromise() {
       if (from) {
         try {
           const promiseResult = await walletService.getAllowance(from);
-          setResult(promiseResult);
-          setError(undefined);
+          setState((prevState) => ({ ...prevState, result: promiseResult, error: undefined, isLoading: false }));
         } catch (e) {
-          setError(e);
+          /* eslint-disable  @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access */
```

Comment:
We can set e as string to avoid this lint warning.

Btw instead of enabling and disabling eslint-rules we can just do
```
// eslint-disable-next-line <eslint rule>
```

And it just works for one line